src/test/java/org/apache/commons/lang3/StringUtilsTest.java

```diff
@@ -3338,6 +3341,95 @@ public void testToRootUpperCase() {
         }
     }
 
+    @Test
+    void extractNumbersFromString() {
+        String input;
+        List<Double> expectedList = new ArrayList<>();
+
+        expectedList.add(123.112);
+        expectedList.add(42.0);
+        input = "Some number 123.112 another is 42.";
+        Assertions.assertEquals(expectedList, StringUtils.extractNumbersFromString(input));
+        expectedList.clear();
+
+        expectedList.add(3.1);
+        expectedList.add(49.0);
+        input = "3.1 pinokio49";
+        Assertions.assertEquals(expectedList, StringUtils.extractNumbersFromString(input));
+        expectedList.clear();
+
+        expectedList.add(1.11);
+        expectedList.add(0.42);
+        input = "Heartbeat : 1.11. pressure is is 0.42";
+        Assertions.assertEquals(expectedList, StringUtils.extractNumbersFromString(input));
+        expectedList.clear();
+
+        expectedList.add(1.11);
+        expectedList.add(0.42);
+        input = "Heartbeat : 1.11. pressure is is 0.42";
+        Assertions.assertEquals(expectedList, StringUtils.extractNumbersFromString(input));
+        expectedList.clear();
+
+        expectedList.add(737.0);
+        expectedList.add(42.23);
+        input = "737 another is 42.23.";
+        Assertions.assertEquals(expectedList, StringUtils.extractNumbersFromString(input));
+        expectedList.clear();
+
+        expectedList.add(737.0);
+        expectedList.add(2193.0);
+        expectedList.add(12.1);
+        expectedList.add(0.11);
+        input = "737 2193, 12.1, 0.11.";
+        Assertions.assertEquals(expectedList, StringUtils.extractNumbersFromString(input));
+        expectedList.clear();
+
+        expectedList.add(737.0);
+        expectedList.add(29.1);
+        expectedList.add(299.0);
+        input = "737. something29.1. 299";
+        Assertions.assertEquals(expectedList, StringUtils.extractNumbersFromString(input));
+        expectedList.clear();
+
+        expectedList.add(737.0);
+        expectedList.add(29.1);
+        expectedList.add(299.0);
+        Assertions.assertThrows(NumberFormatException.class, () -> StringUtils.extractNumbersFromString("737 another is 42.23.32"));
+        expectedList.clear();
+
+        expectedList.add(737.0);
+        expectedList.add(29.1);
+        expectedList.add(299.0);
+        input = "737. something29.1. 299";
+        Assertions.assertEquals(expectedList, StringUtils.extractNumbersFromString(input));
+        expectedList.clear();
```

Comment:
Why is this an error condition? Sentences end in a period.

How would you treat commas in a sentence when in a locale like French that uses the comma as the decimal separator?